### PR TITLE
Fix missing FreeRTOS+TCP files for FreeRTOS console

### PR DESCRIPTION
**Pull request by aggarw13**
Fix the `AFR::freertos_plus_tcp` library module to contain all header files under `freertos_plus_tcp/portable` to be included in FreeRTOS console download. Also, perform hygiene of moving CMake configuration of library from `tests/integration_test` to `libraries/freertos_plus/standard` folder

### CODE REVIEW

**Review by AniruddhaKanhere — DISMISSED** [review]

File: libraries/freertos_plus/standard/freertos_plus_tcp.cmake [hunk 1]
```diff
@@ -0,0 +1,40 @@
+afr_module(NAME "freertos_plus_tcp" INTERNAL)
+
+set(src_dir "${CMAKE_CURRENT_LIST_DIR}/freertos_plus_tcp")
+set(inc_dir "${CMAKE_CURRENT_LIST_DIR}/freertos_plus_tcp/include")
+set(test_dir "${CMAKE_CURRENT_LIST_DIR}")
```
Comment (aggarg) [inline libraries/freertos_plus/standard/freertos_plus_tcp.cmake:5]: The value of test_dir does not seem correct based on what it was earlier.

File: tests/integration_test/CMakeLists.txt [hunk 2]
```diff
@@ -17,62 +17,12 @@ afr_module_dependencies(
...
-    PRIVATE "${test_dir}"
-)
-
-afr_module_dependencies(
-    ${AFR_CURRENT_MODULE}
-    PUBLIC AFR::${AFR_CURRENT_MODULE}::mcu_port
-)
-
-afr_test_module()
+afr_test_module(freertos_tcp)
```
Comment (aggarg) [inline tests/integration_test/CMakeLists.txt:20]: afr_test_module(freertos_tcp_test)

File: libraries/freertos_plus/standard/freertos_plus_tcp.cmake [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        # List of files added to the target so that these are available
+        # in code downloaded from the FreeRTOS console.
+        ${CMAKE_CURRENT_LIST_DIR}/freertos_plus_tcp.cmake
+        ${INCLUDE_HEADER_FILES}
+)
+
+afr_module_include_dirs(
+    ${AFR_CURRENT_MODULE}
+    PUBLIC "${inc_dir}"
+    PRIVATE "${test_dir}"
```
Comment (aggarg) [inline libraries/freertos_plus/standard/freertos_plus_tcp.cmake:34]: Why does FreeRTOS+TCP module depend on test?

File: libraries/freertos_plus/standard/freertos_plus_tcp.cmake [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+set(src_dir "${CMAKE_CURRENT_LIST_DIR}/freertos_plus_tcp")
+set(inc_dir "${CMAKE_CURRENT_LIST_DIR}/freertos_plus_tcp/include")
+set(test_dir "${CMAKE_CURRENT_LIST_DIR}")
+
+# Create a list of all FReeRTOS+TCP headers files in the FreeRTOS+TCP
+# library.
+# The list of header files will be added to metadata required
+# for the FreeRTOS console.
+file(GLOB_RECURSE INCLUDE_HEADER_FILES "${inc_dir}/*.h")
```
Comment (aggarg) [inline libraries/freertos_plus/standard/freertos_plus_tcp.cmake:11]: Do you need to do glob recurse or can it same as MQTT? https://github.com/aws/amazon-freertos/blob/master/libraries/core_mqtt_demo_dependencies.cmake#L12

Also how about more specific variable name FREERTOS_TCP_INCLUDE_HEADER_FILES?

File: libraries/freertos_plus/standard/freertos_plus_tcp.cmake (hunk 3 above)
Comment (aggarw13) [reply libraries/freertos_plus/standard/freertos_plus_tcp.cmake:34]: @AniruddhaKanhere, do you know the reason for it? 
I received build failure in previous commit without the include path dependency to `tests/integration_test`

```
..\..\..\libraries\freertos_plus\standard\freertos_plus_tcp\FreeRTOS_DNS.c(1901): fatal error C1083: Cannot open include file: 'freertos_tcp_test_access_dns_define.h': No such file or directory
```

File: libraries/freertos_plus/standard/freertos_plus_tcp.cmake (hunk 1 above)
Comment (aggarw13) [reply libraries/freertos_plus/standard/freertos_plus_tcp.cmake]: Updated

File: libraries/freertos_plus/standard/freertos_plus_tcp.cmake (hunk 3 above)
Comment (aggarg) [reply libraries/freertos_plus/standard/freertos_plus_tcp.cmake:34]: It should be in the include path for `afr_test_module(freertos_tcp_test)`.

File: libraries/freertos_plus/standard/freertos_plus_tcp.cmake (hunk 4 above)
Comment (aggarw13) [reply libraries/freertos_plus/standard/freertos_plus_tcp.cmake]: I replaced `GLOB_RECURSE` with just `GLOB`, and renamed the macro to `FREERTOS_TCP_HEADER_FILES`

File: libraries/freertos_plus/standard/freertos_plus_tcp.cmake (hunk 3 above)
Comment (aggarg) [reply libraries/freertos_plus/standard/freertos_plus_tcp.cmake:34]: Please ignore the above comment. I realize that the reason is - FreeRTOS_DNS.c includes freertos_tcp_test_access_dns_define.h when the FreeRTOS+TCP is built for test. The file freertos_tcp_test_access_dns_define.h exists in the tests directory and therefore it needs to be in the include path.

Comment (aggarg) [reply libraries/freertos_plus/standard/freertos_plus_tcp.cmake:34]: To make it clear that test dir inclusion is only needed when building for test, I'd suggest the following:

```
afr_module_include_dirs(
    ${AFR_CURRENT_MODULE}
    PUBLIC "${inc_dir}"
)
if(AFR_ENABLE_TESTS)
    afr_module_include_dirs(
        ${AFR_CURRENT_MODULE}
        PRIVATE
            "${AFR_TESTS_DIR}/integration_test"
    )
endif()
```

**Review by aggarg — APPROVED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Pull request merged**